File: back-end/internal/domain/monthlysummary/monthly_summary.go
package monthlysummary

import (
	"context"
	"time"
)

// MonthlySummary は対象月の収入、固定費、出費、残額の正計算結果を表す。
type MonthlySummary struct {
	Month           string
	TotalIncome     int
	AvailableIncome int
	ReservedIncome  int
	FixedCostTotal  int
	ExpenseTotal    int
	RemainingAmount int
	ActualBalance   int
}

// Repository は月次サマリーに必要なDB集計を行う。
type Repository interface {
	Get(ctx context.Context, userID, month string, start, end time.Time) (MonthlySummary, error)
}

File: front-end/src/domains/monthlySummary.ts
/**
 * 月次サマリーdomain model。
 */
export type MonthlySummary = {
  /** 実収支 */
  readonly actualBalance: number;
  /** 使える収入 */
  readonly availableIncome: number;
  /** 出費合計 */
  readonly expenseTotal: number;
  /** 固定費合計 */
  readonly fixedCostTotal: number;
  /** YYYY-MM形式の対象月 */
  readonly month: string;
  /** 今月の残り予算 */
  readonly remainingAmount: number;
  /** 貯める収入 */
  readonly reservedIncome: number;
  /** 全収入 */
  readonly totalIncome: number;
};
